src/plugins/kdb/ldap/libkdb_ldap/ldap_principal2.c

```diff
@@ -1375,64 +1482,207 @@ krb5_read_tkt_policy(krb5_context context, krb5_ldap_context *ldap_context,
     return st;
 }
 
-krb5_error_code
-krb5_decode_krbsecretkey(krb5_context context, krb5_db_entry *entries,
-                         struct berval **bvalues,
-                         krb5_tl_data *userinfo_tl_data, krb5_kvno *mkvno)
+struct key_entry {
```

Comment:
The name "key_entry" is deceptive since this structure holds multiple key data entries.  This should probably be named "struct keyset" after KrbKeySet, and the arguments which refer to it should use "keyset" rather than "key" as nouns (e.g. "keysets_out" instead of "keys_out" in decode_keys() below).